### DISCUSSION TITLE
MongoDB with Panache and Kerberos

### DISCUSSION THREAD

**Discussion by doogle-oss** [quarkusio/quarkus#23521] [Q&A]
Hi team,
We are looking at moving to mongodb with kerberos connectivity using Quarkus. Can someone help on how to configure.
Right now we do not have issue as the database is open but now the team is adding kerberos protection.
Do advise on this.

[a comment by an automated account is not shown]

**Reply by doogle-oss** [reply]
A quarkus insight will help as well

**Comment by loicmathieu** [comment]
I never used Kerberos authentication myself, it's an enterprise feature.

According to the documentation, client side, you should configure:
```
quarkus.mongodb.credentials.auth-mechanism=GSSAPI
#The Kerberos user name, including the realm, e.g. "user1@MYREALM.ME"
quarkus.mongodb.credentials.username=<your_kerberos_user>
```

Some Java system properties must be defined as documented in the MongoDB documentation: http://mongodb.github.io/mongo-java-driver/3.0/driver/reference/connecting/authenticating/#kerberos-gssapi

**Reply by doogle-oss** [reply]
Thanks let me try it out
